docker/container.go

```diff
@@ -209,10 +251,28 @@ func (c *Container) OutOfSync() (bool, error) {
 		return false, err
 	}
 
-	return info.Config.Labels[HASH.Str()] != project.GetServiceHash(c.service), nil
+	logrus.WithFields(logrus.Fields{
+		"info.Config.Labels[HASH.Str()]": info.Config.Labels[HASH.Str()],
+		"c.getHash()":                    c.getHash(),
+	}).Debug("Comparing old and must-be hash")
+	return info.Config.Labels[HASH.Str()] != c.getHash(), nil
+}
+
+func (c *Container) getHash() string {
+	return project.GetServiceHash(c.service.Name(), c.service.Config())
 }
 
-func (c *Container) createContainer(imageName string) (*dockerclient.APIContainers, error) {
+func volumeBinds(volumes map[string]struct{}, container *dockerclient.Container) []string {
```

Comment:
The python code looks at external: https://github.com/docker/compose/blob/5ed9f9/compose/service.py#L926  Is that covered here?